### ISSUE TITLE
Workflows branch: running a workchain with runner yields kiwi exception

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#1089]
Simple example of trying to run the `PwBaseWorkChain` with `aiida.work.launch.run` results in the following exception:

```
1/30/2018 03:15:48 PM, aiida.aiida.orm.implementation.general.calculation.work.WorkCalculation: [REPORT] [213|PwBaseWorkChain|run_calculation]: launching PwCalculation<216> iteration #1
Traceback (most recent call last):
 File "/home/seb/.virtualenvs/aiida_dev/bin/launch_workflow_pw_base", line 11, in <module>
   load_entry_point('aiida-quantumespresso', 'console_scripts', 'launch_workflow_pw_base')()
 File "/home/seb/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
   return self.main(*args, **kwargs)
 File "/home/seb/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/click/core.py", line 697, in main
   rv = self.invoke(ctx)
 File "/home/seb/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
   return ctx.invoke(self.callback, **ctx.params)
 File "/home/seb/.virtualenvs/aiida_dev/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
   return callback(*args, **kwargs)
 File "/home/seb/code/aiida/env/dev/aiida-quantumespresso/aiida_quantumespresso/cli/workflows/pw/base.py", line 58, in launch
   run(PwBaseWorkChain, **inputs)
 File "/home/seb/code/aiida/env/dev/aiida-core/aiida/work/launch.py", line 48, in run
   return runner.run(process, *args, **inputs)
 File "/home/seb/code/aiida/env/dev/aiida-core/aiida/work/runners.py", line 184, in run
   return process.execute()
 File "/home/seb/code/plum/plum/process.py", line 440, in execute
   return Executor(return_on_idle).execute(self)
 File "/home/seb/code/plum/plum/process.py", line 119, in execute
   return self._future.result()
 File "/home/seb/code/kiwi/kiwipy/futures.py", line 41, in result
   raise InvalidStateError("Not done yet so no result")
kiwipy.futures.InvalidStateError: Not done yet so no result
```

**Comment by muhrin** [comment]
Thanks Seb.  This is caused by kiwipy issue https://github.com/muhrin/kiwipy/issues/1

**Comment by muhrin** [comment]
@sphuber Aiight, it took a while but I have what I think is working solution.  Could you please have a gander at:

https://github.com/muhrin/aiida_core/tree/fix_1089_workchain_yield_without_result

Now, the tests are failing big time:

https://travis-ci.org/muhrin/aiida_core/builds/335775475

Because I changed the travis-ci to use the requirements.txt file instead of what's in setup.py.  This way we can always have a working version even if I need to refer to a specific plum or kiwi commit combination.  The reasons for the tests are:
1) Documentation requirements are fucked - would you be able to have a look? and
2) The calculations at the end get stuck in the 'TOSUBMIT' state.  Do you have any ideas?

**Comment by sphuber** [comment]
Aye cap'n. I will give it a little lookyloo tomorrow morning first thing

**Comment by muhrin** [comment]
Fixed by eb79b00cd64b502322bc3f3d6c717236b66b553c
